### PR TITLE
Snippet scanning framework

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

Add a library that sets up:

- A trait for snippet extractors.
- Types for those extractors to use.

## Acceptance criteria

A library is available for implementing extractors.

## Testing plan

No actual implementation here, but the few invariants we can't easily enforce in the type system are tested.

## Risks

None

## References

Implements https://fossa.atlassian.net/browse/ANE-1116

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).

### CODE REVIEW

File: snippets/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+/// ```
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[non_exhaustive]
+pub enum Method {
+    /// Generated from the text with the specified normalizations applied.
+    Normalized(Transforms),
+
+    /// Generated from the text as written.
+    ///
+    /// ```ignore
```
Comment (spatten) [inline snippets/src/lib.rs:267]: nit: IMO the code-block here doesn't add anything to the explanation

**Review by spatten — APPROVED** [review]
This looks good to me! Love the detailed comments!

File: snippets/src/lib.rs (hunk 1 above)
Comment (jssblck) [reply snippets/src/lib.rs:267]: Ha, good catch. That was originally meant to be compared more directly with the `Transforms` examples, but that's not really the case anymore. Will remove!

Comment (jssblck) [reply snippets/src/lib.rs:267]: Actually, I've made a few changes to this on my follow up branch, will just include this change there too and merge this branch as-is.

**Pull request merged**